complex/ft_config.c

```diff
@@ -40,8 +41,38 @@
 #define FT_MODE_ALL	/*FI_CONTEXT |*/ FI_LOCAL_MR /*| FI_MSG_PREFIX*/
 #define FT_MODE_NONE	~0ULL
 
+#define STRINGIFY(x) #x
+#define STREQ(STR1, STR2) !strncmp(STR1, STR2, strlen(STR2))
+
+#define TEST_ENUM_SET_N_RETURN(str, enum_val, type, key)	\
+	do {							\
+		if (STREQ(str, STRINGIFY(enum_val))) {		\
+			*(type *)(key) = enum_val;		\
+			return 0;				\
+		}						\
+	} while (0)
+
+#define TEST_SET_N_RETURN(str, val_str, val, type, key)	\
+	do {						\
+		if (STREQ(str, val_str)) {		\
+			*(type *)(key) = val;		\
+			return 0;			\
+		}					\
+	} while (0)
+
+#define ftset_offsetof(field) offsetof(struct ft_set, field)
+#define ftset_sizeof(field) sizeof(((struct ft_set *)0)->field)
+
+#define FT_SET_MAX 11
```

Comment:
> > +#define FT_SET_MAX 11
> 
> Dropping this would have the magic number 11 at couple of places in the
> code. So keep it?

Why can't you declare the array with just [] and use sizeof?